cpython-windows/build.py

```diff
@@ -233,26 +237,34 @@ def find_vs_path(path, msvc_version):
     else:
         raise ValueError(f"unsupported Visual Studio version: {msvc_version}")
 
-    p = subprocess.check_output(
-        [
-            str(vswhere),
-            # Visual Studio 2019.
-            "-version",
-            version,
-            "-property",
-            "installationPath",
-            "-products",
-            "*",
-        ]
-    )
+    cmdline = [
+        str(vswhere),
+        # Visual Studio 2019.
+        "-version",
+        version,
+        "-property",
+        "installationPath",
+        "-prerelease",
+        "-products",
+        "*",
+    ]
+    p = subprocess.check_output(cmdline)
+    if not p:
+        cmd = " ".join('"%s"' % i for i in cmdline)
```

Comment:
Is there a reason to quote each item in the invocation? I feel like it'd probably be more readable without.

Reply:
I was trying to balance between copy-pasteability and complexity. The first argument is typically going to have multiple spaces in it :/

"C:\Program Files (x86)\Microsoft\Visual Studio 2019\".. le sigh :( Quoting only the ones that have spaces in is doable but kind of hard to read:

```
cmd = " ".join([
  '"%s"' if ' ' in i else i
  for i in cmdline
  ])

Reply:
[`shlex.join`](https://docs.python.org/3/library/shlex.html#shlex.join) may be what you're looking for. The quoting isn't designed for non-POSIX shells but should do the job for pretty-printing.